oar/core/shipment.py

```diff
@@ -257,6 +261,149 @@ def get_status(self) -> str:
+            
+            stage_work = [w for w in all_work if hasattr(w, 'stage') and w.stage == stage_name]
+            
+            if not stage_work:
+                # Log all work for debugging
+                logger.debug(f"All pipeline work: {[(w.name, getattr(w, 'stage', None), w.status) for w in all_work]}")
+                raise GitLabMergeRequestException(
+                    f"Stage '{stage_name}' not found in pipeline. Available stages: {all_stages}")
+                
+            # Return status from first job in stage (GitLab handles failed status automatically)
+            job = stage_work[0]
+            status = job.status
+            logger.debug(f"Stage '{stage_name}' status: {status}")
+            
+            return {
+                'status': status,
+                'job': job
+            }
+        except Exception as e:
+            logger.error(f"Error getting stage status: {str(e)}")
+            raise
+
+    def is_stage_release_success(self) -> bool:
+        """Check if the stage-release-triggers stage has succeeded
+        
+        Returns:
+            bool: True if stage status is 'success', False otherwise
+            
+        Raises:
+            GitLabMergeRequestException: If unable to get stage status
+        """
+        try:
+            stage_info = self.get_stage_release_info()
+            if stage_info['status'] == 'success':
+                return True
+                
+            logger.error(f"Stage release failed with status {stage_info['status']}")
+            logger.error(f"Job details: {stage_info['job'].pformat()}")
+            return False
+        except Exception as e:
+            logger.error(f"Failed to check stage release status: {str(e)}")
+            raise GitLabMergeRequestException(f"Failed to check stage release status: {str(e)}") from e
+
+    def get_stage_release_info(self) -> dict:
+        """Get detailed info about the 'stage-release-triggers' stage from latest pipeline
+        
+        Returns:
+            dict: {
+                'status': overall stage status (e.g. 'running', 'success', 'failed'),
+                'jobs': [list of failed jobs if stage failed]  # Each job includes pipeline_id
+            }
+            
+        Raises:
+            GitLabMergeRequestException: If stage not found or pipeline error
+        """
+        stage_name = "stage-release-triggers"
+        try:
+            pipeline = self._get_latest_pipeline()
+            
+            
+            try:
+                stage_status = self._get_stage_info_from_pipeline(stage_name, pipeline)
+                logger.info(f"Stage '{stage_name}' status: {stage_status}")
+                return stage_status
+            except GitLabMergeRequestException as e:
+                logger.error(f"Stage '{stage_name}' not found in pipeline {pipeline.id}")
+                return {
+                    'status': 'not_found',
+                    'jobs': []
```

Comment:
I think `job` and `jobs` are mixed. `_get_stage_info_from_pipeline` returns `job` but here is `job`. Also in the method docs.

Reply:
Fixed